File: graphql-dgs-codegen-core/src/integTest/java/com/netflix/graphql/dgs/codegen/java/testcases/dataclasses/dataClassWithBooleanField/expected/types/RequiredTestType.java
package com.netflix.graphql.dgs.codegen.java.testcases.dataclasses.dataClassWithBooleanField.expected.types;

import java.lang.Object;
import java.lang.Override;
import java.lang.String;
import java.util.Objects;

public class RequiredTestType {
  private boolean isRequired;

  public RequiredTestType() {
  }

  public RequiredTestType(boolean isRequired) {
    this.isRequired = isRequired;
  }

  public boolean getIsRequired() {
    return isRequired;
  }

  public void setIsRequired(boolean isRequired) {
    this.isRequired = isRequired;
  }

  @Override
  public String toString() {
    return "RequiredTestType{isRequired='" + isRequired + "'}";
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) return false;
    RequiredTestType that = (RequiredTestType) o;
    return isRequired == that.isRequired;
  }

  @Override
  public int hashCode() {
    return Objects.hash(isRequired);
  }

  public static Builder newBuilder() {
    return new Builder();
  }

  public static class Builder {
    private boolean isRequired;

    public RequiredTestType build() {
      RequiredTestType result = new RequiredTestType();
      result.isRequired = this.isRequired;
      return result;
    }

    public Builder isRequired(boolean isRequired) {
      this.isRequired = isRequired;
      return this;
    }
  }
}

File: graphql-dgs-codegen-core/src/integTest/kotlin/com/netflix/graphql/dgs/codegen/cases/dataClassWithBooleanField/expected/types/RequiredTestType.kt
package com.netflix.graphql.dgs.codegen.cases.dataClassWithBooleanField.expected.types

import com.fasterxml.jackson.`annotation`.JsonIgnoreProperties
import com.fasterxml.jackson.`annotation`.JsonProperty
import com.fasterxml.jackson.`annotation`.JsonTypeInfo
import com.fasterxml.jackson.databind.`annotation`.JsonDeserialize
import com.fasterxml.jackson.databind.`annotation`.JsonPOJOBuilder
import com.netflix.graphql.dgs.codegen.cases.dataClassWithBooleanField.expected.Generated
import java.lang.IllegalStateException
import java.util.Objects
import kotlin.Any
import kotlin.Boolean
import kotlin.Int
import kotlin.String
import kotlin.jvm.JvmName

@Generated
@JsonTypeInfo(use = JsonTypeInfo.Id.NONE)
@JsonDeserialize(builder = RequiredTestType.Builder::class)
public class RequiredTestType(
  isRequired: () -> Boolean = isRequiredDefault,
) {
  private val __isRequired: () -> Boolean = isRequired

  @get:JvmName("getIsRequired")
  public val isRequired: Boolean
    get() = __isRequired.invoke()

  override fun equals(other: Any?): Boolean {
    if (this === other) return true
    if (other !is RequiredTestType) return false
    return (__isRequired === isRequiredDefault) == (other.__isRequired === isRequiredDefault) &&
        (__isRequired === isRequiredDefault || Objects.equals(isRequired, other.isRequired))
  }

  override fun hashCode(): Int = Objects.hash(if (__isRequired === isRequiredDefault)
      isRequiredDefault else isRequired)

  override fun toString(): String = listOfNotNull(if (__isRequired === isRequiredDefault) null else
      "isRequired=" + isRequired).joinToString(prefix = "RequiredTestType(", postfix = ")")

  @Generated
  public companion object {
    private val isRequiredDefault: () -> Boolean = 
        { throw IllegalStateException("Field `isRequired` was not requested") }
  }

  @Generated
  @JsonPOJOBuilder
  @JsonIgnoreProperties("__typename")
  public class Builder {
    private var isRequired: () -> Boolean = isRequiredDefault

    @JsonProperty("isRequired")
    public fun withIsRequired(isRequired: Boolean): Builder = this.apply {
      this.isRequired = { isRequired }
    }

    public fun build(): RequiredTestType = RequiredTestType(
      isRequired = isRequired,
    )
  }
}
